music21/stream/base.py

```diff
@@ -5287,11 +5289,33 @@ def _transposeByInstrument(
             trans = inst.transposition
             if reverse:
                 trans = trans.reverse()
+
+            displayStatuses = {}
+            if transposeKeySignature and displayStatusesAreSet:
```

Comment:
See notes in "conversation" about this.

Reply:
I have followed your suggestions in the "conversation": makeNotation.py now has a `saveAccidentalDisplayStatus` generator/contextmanager that `_transposeByInstrument` uses if it was passed `preserveAccidentalDisplay=True` (`toWrittenPitch` and `toSoundingPitch` are also passed `preserveAccidentalDisplay`, which they pass to `_transposeByInstrument`).  I couldn't use the `lambda`, though, because pylint complained about using `focus` in the `lambda` inside a loop where `focus` was defined (or something).

I had not heard of contextmanager before; very cool, and fun to learn something new.

Reply:
Done.